packages/react/src/components/Grid/__tests__/Grid-test.js

```diff
@@ -7,9 +7,71 @@
 
 import { render } from '@testing-library/react';
 import React from 'react';
-import { Grid } from '../';
+import { Grid, FlexGrid } from '../';
+
+describe('FlexGrid', () => {
+  it('should support a custom element as the root node', () => {
+    const { container } = render(<FlexGrid as="section" />);
+    expect(container.firstChild.tagName).toBe('SECTION');
+  });
+
+  it('should include a custom className', () => {
+    const { container } = render(<FlexGrid className="test" />);
+    expect(container.firstChild).toHaveClass('test');
+  });
+
+  it('should pass un-used props to the top-level node that is rendered', () => {
+    const { container } = render(<FlexGrid id="test" />);
+    expect(container.firstChild).toHaveAttribute('id', 'test');
+  });
+
+  it('should render `children` that are given', () => {
+    const { container } = render(
+      <FlexGrid>
+        <span id="test">Test</span>
+      </FlexGrid>
+    );
+    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+    const testNode = container.querySelector('#test');
+    expect(testNode).toBeInstanceOf(HTMLElement);
+  });
+
+  it('should support setting the condensed class through the `condensed` prop', () => {
+    const { container } = render(<FlexGrid condensed />);
+    expect(container.firstChild.className).toEqual(
+      expect.stringContaining('grid--condensed')
+    );
+  });
+
+  it('should support setting the full-width class through the `fullWidth` prop', () => {
+    const { container } = render(<FlexGrid fullWidth />);
+    expect(container.firstChild.className).toEqual(
+      expect.stringContaining('grid--full-width')
+    );
+  });
+});
 
 describe('Grid', () => {
+  let Grid;
+  let cleanup;
+  let render;
+  let screen;
+
+  beforeEach(() => {
+    jest.resetModules();
+    const FeatureFlags = require('@carbon/feature-flags');
+    FeatureFlags.enable('enable-css-grid');
+
+    cleanup = require('@testing-library/react/pure').cleanup;
+    render = require('@testing-library/react/pure').render;
+    screen = require('@testing-library/react/pure').screen;
+    Grid = require('../Grid').Grid;
```

Comment:
Hey I'm curious, is this a new pattern we need to follow in certain situations? I know there's been some changes like this in best practices for `@testing-library` but haven't seen this before.

Reply:
I actually pulled this code from the Column-test file when looking for an example of how we were enabling the feature flag within tests.